pkg/store/bucket.go

```diff
@@ -1585,11 +1632,22 @@ func (r *bucketChunkReader) addPreload(id uint64) error {
 }
 
 // preload all added chunk IDs. Must be called before the first call to Chunk is made.
-func (r *bucketChunkReader) preload() error {
+func (r *bucketChunkReader) preload(samplesLimiter *Limiter) error {
 	const maxChunkSize = 16000
```

Comment:
we could perhaps move this to the top along with `maxSamplesPerChunk`?

Reply:
No necessarily as it only is used here